pkgs/collection/test/extensions_test.dart

```diff
@@ -1122,6 +1122,87 @@ void main() {
         });
       });
     });
+    group('of MapEntry', () {
+      group('.whereKey', () {
+        test('empty', () {
+          expect(<String, int>{}.entries.whereKey(unreachable), isEmpty);
+        });
+        test('single', () {
+          expect({'a': 1}.entries.whereKey((k) => k == 'a').toMap(), {'a': 1});
```

Comment:
Don't use `toMap` in the test of `whereKey`. If there is a bug in `toMap`, it might hide a bug in `whereKey`.

Check against `[MapEntry('a', 1)]` instead.

I'd even say to not use `map.entries` as the source. Just have something like:
```dart
Iterable<MapEntry<K, V>> asIterable<K, V>(Iterable<MapEntry<K, V>> entries) => entries;
// ...
  expect(asIterable([MapEntry('a', 1)]).whereKey((k) => k == 'a'), [MapEntry('a', 1)]);
```
The `asIterable` is only there to change the static type away from `List<MapEntry>`, in case that matters. (It shouldn't, but this is testing, not assuming things work.)

Reply:
Done, I don't mind, but we are effectively testing the behavior of the Dart SDK.

So I'm not really sure this does much other than make the tests harder to read :rofl: